src/jinja2/environment.py

```diff
@@ -1120,7 +1120,13 @@ def make_module_async(self, vars=None, shared=False, locals=None):
         )
 
     @internalcode
-    def _get_default_module(self):
+    def _get_default_module(self, ctx=None):
+        if ctx is not None:
```

Comment:
It would be helpful to have a comment here describing why this happens. "If a context is passed in, that means {explanation}. If there are new globals, we need to create a new moudle because {explanation}."